kernel/src/loader/mod.rs

```diff
@@ -144,6 +148,76 @@ where
     Ok(GuestAddress(ehdr.e_entry as usize))
 }
 
+#[cfg(target_arch = "aarch64")]
+pub fn load_kernel<F>(
+    guest_mem: &GuestMemory,
+    kernel_image: &mut F,
+    start_address: usize,
+) -> Result<GuestAddress>
+where
+    F: Read + Seek,
+{
+    /* Kernel boot protocol is specified in the kernel docs
+     * Documentation/arm/Booting and Documentation/arm64/booting.txt
+     */
+    const AARCH64_KERNEL_LOAD_ADDR: usize = 0x80000;
+    const AARCH64_MAGIC_NUMBER: u32 = 0x644d5241;
+    let mut kernel_load_offset = AARCH64_KERNEL_LOAD_ADDR;
+
+    /* Look for the magic number inside the elf header. */
+    let magic_offset_header = 2 * mem::size_of::<u32>() + 6 * mem::size_of::<u64>(); // This should total 56.
```

Comment:
Any reason why we do not make the `magic_number_offset` a constant? Do we need to use `mem::size_of` instead of using 4 and 8 (other than say code readability maybe?)?

Reply:
You can have the best of both worlds by writing
```rust
const MAGIC_OFFSET_HEADER: usize =  2 * mem::size_of::<u32>() + 6 * mem::size_of::<u64>(); 
```
Same goes for `text_offset`, so you can keep both the verbose computation and move them up here with the other constants.

Reply:
I think we can just use  immediate "4" and "8" for the field size. Per arm64 booting protocol, the field size if fixed to u32 or u64 which have defined the field length to a specific value.